### PR TITLE
Python dependency upgrades

### PR DESCRIPTION
**Pull request by wlonk**
Update all Python dependencies.

### CODE REVIEW

File: metashare/api/validators.py [hunk 1]
```diff
@@ -13,21 +13,23 @@ def process_field_name(self, field_name):
...
+        # field names => field sources
+        sources = [serializer.fields[field_name].source for field_name in self.fields]
+
         # If this is an update, then any unprovided field should
-        # have its value set based on the existing instance attribute.
-        if self.instance is not None:
-            for field_name in self.fields:
-                if field_name not in attrs:
-                    attrs[field_name] = getattr(self.instance, field_name)
+        # have it's value set based on the existing instance attribute.
```
Comment (jgerigmeyer) [inline metashare/api/validators.py]:
```suggestion
        # have its value set based on the existing instance attribute.
```

File: metashare/api/validators.py [hunk 2]
```diff
@@ -13,21 +13,23 @@ def process_field_name(self, field_name):
...
+        if serializer.instance is not None:
+            for source in sources:
+                if source not in attrs:
+                    attrs[source] = getattr(serializer.instance, source)
 
         # Determine the filter keyword arguments and filter the queryset.
         filter_kwargs = {
-            self.process_field_name(field_name): attrs[field_name]
-            for field_name in self.fields
```
Comment (jgerigmeyer) [inline metashare/api/validators.py]: Curious what prompted these changes?

File: metashare/api/validators.py (hunk 1 above)
Comment (wlonk) [reply metashare/api/validators.py]: This was copied verbatim from the relevant bit of DRF source, so I should push that fix upstream too :)

File: metashare/api/validators.py (hunk 2 above)
Comment (wlonk) [reply metashare/api/validators.py]: Changes in DRF.

File: metashare/api/validators.py (hunk 1 above)
Comment (jgerigmeyer) [reply metashare/api/validators.py]: Maybe add a comment that this is copied from elsewhere?

**Pull request merged**